plugins/Overlay/javascripts/Piwik_Overlay.js

```diff
@@ -158,6 +164,41 @@ var Piwik_Overlay = (function () {
         updateComesFromInsideFrame = false;
     }
 
+    function handleApiRequests() {
+        window.addEventListener("message", function (event) {
+            if (event.origin !== iframeOrigin) {
+                return;
+            }
+
+            var strData = event.data.split(':', 3);
+            if (strData[0] !== 'overlay.call') {
+                return;
+            }
+
+            var requestId = strData[1];
+            var url = decodeURIComponent(strData[2]);
+
+            var params = broadcast.getValuesFromUrl(url);
+
```

Comment:
Can we here only allow  specific API methods needed for overlay to increase the security?